gaeguli/pipeline.c

```diff
@@ -904,13 +904,18 @@ gaeguli_pipeline_suggest_buffer_size_for_target (GaeguliPipeline * self,
 GaeguliTarget *
 gaeguli_pipeline_add_srt_target_full (GaeguliPipeline * self,
     GaeguliVideoCodec codec, guint bitrate, const gchar * uri,
-    const gchar * username, GError ** error)
+    const gchar * username, gboolean is_record_target,
+    const gchar * location, GError ** error)
```

Comment:
it leads to ambiguity since `uri` and `is_record_target=TRUE` are mutually exclusive.

I'd prefer having separate functions to create each target object.
How about having a static common function like these functions?

```
static GaeguliTarget *gaeguli_pipeline_add_target();

GaeguliTarget *gaeguli_pipeline_add_srt_target_full()
{
  gaeguli_pipeline_add_target (uri, FALSE);
}
GaeguliTarget *gaeguli_pipeline_add_recording_target_full()
{
  gaeguli_pipeline_add_target (NULL, TRUE);
}
```